### ISSUE TITLE
1.18.2 Crash Involving Tool's Complement

### ISSUE THREAD

**Issue by RConTRS** [Fourmisain/AxesAreWeapons#12]
I Was Breaking Blocks Using The Diamond Hammer, After 3 Breaks The Game Crashed, Looks Like It's NullPointerException
I Got NotEnoughCrashes And It Mentioned AxesAreWeapons, I Have No Clue So Here's The Only Crash Report Generated

[crash-2023-01-21_16.52.54-client.txt](https://github.com/Fourmisain/AxesAreWeapons/files/10472313/crash-2023-01-21_16.52.54-client.txt)

**Comment by Fourmisain** [comment]
This one again!
This issue has had a long history (#7, #10), I could never reproduce it and it made no sense to me to this day.
Good news is that since this crash is on Forge 1.18, I think I can fix it the same way I fixed it for Forge 1.19.

Even better news is that I think I got an idea of why this is happening in the first place!
This might have been a threading issue this whole time!
That makes sense given my code and it would explain why it would only happen rarely and randomly.
The weird thing about that is that vanilla only ever accesses the methods on the main thread, so there has to be a mod calling it off-thread.

Are you maybe running a mod like WTHIT, megane or similar which is showing you if a block can be mined?
